java-design-patterns/src/main/java/org/example/Singleton/SingletonCache.java

```diff
@@ -0,0 +1,47 @@
+package org.example.Singleton;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SingletonCache<K,V> implements Serializable, Cloneable {
+
+    private static SingletonCache singletonCache = null;
+
+    private Map<K, V> cache = new HashMap<K, V>();
+
+    public static <K,V> SingletonCache<K, V> getInstance(){
+        if (singletonCache == null){
+            synchronized (SingletonCache.class){
+                if (singletonCache == null){
+                    singletonCache = new SingletonCache();
+                }
+            }
+        }
+        return singletonCache;
+    }
+
+    /**
+     * To prevent Object to be created by serialization / deserialization
+     */
+    protected Object readResolve() { return singletonCache; }
```

Comment:
nit: this can be null if not initialized. may be `return getInstance()`

Reply:
used getInstance() instead of singletonCache variable.